pkg/volume/util/subpath/subpath_linux.go

```diff
@@ -315,7 +315,23 @@ func doCleanSubPath(mounter mount.Interface, fullContainerDirPath, subPathIndex
 	klog.V(4).Infof("Cleaning up subpath mounts for subpath %v", subPathIndex)
 	fullSubPath := filepath.Join(fullContainerDirPath, subPathIndex)
 
-	if err := mount.CleanupMountPoint(fullSubPath, mounter, true); err != nil {
+	quietUnmounter, ok := mounter.(mount.MounterQuietUnmounter)
+	var err error
+	if ok {
+		// Avoid expensive mount point checks when umount error output could be
+		// checked instead. Every bind mount adds an entry to the mount table and
+		// the larger the mount table grows the more harder it is to get a
+		// consistent read of it to check it for mount points. This phenomenon is
+		// not specific to subPaths necessarily but it is especially important to
+		// optimize subPath clean up because a single unmount volume operation
+		// could entail dozens of subPath clean ups and if any one of them fails
+		// due to consistent read error then the entire operation must be retried.
+		// See #105642.
+		err = mount.CleanupMountPointQuiet(fullSubPath, quietUnmounter)
```

Comment:
how about just use mount.CleanupMountPoint(fullSubPath, mounter, false)? 
CleanupMountPointQuiet removed checking mountpoint, how -quite option helps here?

Reply:
I reworked the PR so that neither extensive or non-extensive mount point check is needed, as long as we check "not mounted" error from umount. Let me know if this makes more sense than wat i had before:

extensiveMountPointCheck is ignored because:
// assuming that the Linux implementation of Unmount will not error if the
// mountPath is not actually a mount point, there is no need to do a mount
// point check here.

https://github.com/kubernetes/kubernetes/pull/109117/files#diff-2a2377a9ec20eba3d64e9849afc49d5f91e0def59e4ec106131bbed7a5879debR66-R69